pkg/util/provider/machinecontroller/machine_safety.go

```diff
@@ -198,6 +198,19 @@ func (c *controller) AnnotateNodesUnmanagedByMCM(ctx context.Context) (machineut
 					return machineutils.MediumRetry, err
 				}
 			}
+		} else {
+			// FIX: Remove NotManagedByMCM for managed nodes https://github.com/gardener/machine-controller-manager/issues/863
+			_, hasAnnot := node.Annotations[machineutils.NotManagedByMCM]
+			if !hasAnnot {
+				continue
+			}
+			klog.V(3).Infof("Removing NotManagedByMCM annotation from Node %q", node.Name)
+			nodeCopy := node.DeepCopy()
+			delete(nodeCopy.Annotations, machineutils.NotManagedByMCM)
+			_, err := c.targetCoreClient.CoreV1().Nodes().Update(ctx, nodeCopy, metav1.UpdateOptions{})
+			if err != nil {
+				klog.Errorf("Failed to remove NotManagedByMCM from Node %q due to: %s", node.Name, err)
```

Comment:
could you also add similar error message when applying annotation fails due to Update failure.
Also what retry period will be used on this failure?

Reply:
Just to clarify this. There is already an existing error message in `updateNodeWithAnnotation`  

```
	klog.Errorf("Couldn't patch the node %q , Error: %s", node.Name, err)
```
Do you want me to change this ?

Reply:
If the method is not used anywhere other than these annotation, then we can replace it with your version of method..  the idea is to use same method for updating nodes in Annotating/deAnnotating nodes

Reply:
Ok, harmonized use of both add/remove to use same method and fixed logging